### PR TITLE
Feature/users verifications

### PR DESCRIPTION
**Pull request by bestwebua**
Complete users verification (by email) feature.

### CODE REVIEW

File: app/concepts/documentations/operation/show.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@ class Show < Trailblazer::Operation
     fail :file_not_found, fail_fast: true
```
Comment (polykoi) [inline app/concepts/documentations/operation/show.rb:4]: indentation?

File: app/concepts/v1/lib/representer/verification_errors_serializer.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+module V1::Lib::Representer
+  class VerificationErrorsSerializer < JSONAPI::Serializable::Error
+    def initialize(exposures)
```
Comment (polykoi) [inline app/concepts/v1/lib/representer/verification_errors_serializer.rb:5]: change errors serialization to match JSON:API conventions

File: config/credentials.yml.enc [hunk 3]
```diff
@@ -1 +1 @@
-ZdPVSvSjZyJHMJNiVL1TyjQ14DdTL9/giU8nh5LL5wElhdZe8vxhCVNxFn6cOg5KKHM2oBZ8FLkdmmPr1yAHHn2weoiGZpwfodhzgtd9pFmawcBPOLhQ11vcFb8RjJnkRcXIYvCePpRK7tebMFY5Iuah9wGVbEhoU+Pm2SNvNh5M+zmcnoIIAd2bgdaHM9lVy4wcpp4Sp6qyrx0BTDwiHpbzKhEOK5zCqdEttQz0sUwzpURfmOUBdimK67D/a5qlwwUJ/iilQzLgukz+hKOx5lOXIyfN8tfTe/ErAEVjzg6oSyxsBFrgQ9EQJb4DX3z9HijNdwRAHWebA8yHZTyaL8nZHUx0SM4zyLiENM257jmltPoEkSwL8bYW6hSCdAWKs4wZp/naab38PgmK0ER0suWrI/j/LU2QZk+D--VuDPyS5KZ8E0BKIr--3NNUEHpcXbsh9Mcf/GiuMA==
\ No newline at end of file
+PwAJgM4hhd3BKEYMWeQ3MLm7RcV9j0tobbSzzy9wpTW46z3rRtOq4vzjcBKfTIl5p38HxKzsi2zsY+yFLszaF8BP2+9a2BZ41wvJ4VKxNb62B4iaS/yX87imkfDogxpt/2k5J4jx03oGM4By28JUWMNYCHsZ19c0gjEdykYtTTS/5NARAHx1nPHnLXMcqB1alhIy6XZ78Kw3JXMRGurxguqHftHDn5Ue4RH1I3d3bhXBtwMsQT9Vytd4oWiqO8h7bO0itkXp5LqjAy5UTh9UiyFUuq0SJFLxU5Xbd9hHFmZ/s0WM7MK8n+lervgtzuBsoINA6Ja9+26YVQaxFnHB2u1To33/Zmqzzne8tOnIe4vCZeLUlrwYvEvQGiEsmqaWUn02AKGi2x8+OUwFEO3Q7YlgC6FjpsWvQNaRlvAa6fK+xMDmksNx2ugf0cOZGduBifWNWELLwpw=--DimBRbJTnmba3PhB--FpCPc8i9VEIvXlHpB2nHxw==
```
Comment (polykoi) [inline config/credentials.yml.enc:1]: newline at eof

File: config/environments/development.rb [hunk 4]
```diff
@@ -52,3 +59,9 @@
   # routes, locales, etc. This feature depends on the listen gem.
   config.file_watcher = ActiveSupport::EventedFileUpdateChecker
 end
+
+Sidekiq::Cron::Job.create(
```
Comment (polykoi) [inline config/environments/development.rb]: move to initializers

Comment (polykoi) [reply config/environments/development.rb]: is cron jobs persisted over app restart? if they do, check for existence before creatig

File: spec/concepts/documentations/operation/show_spec.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
+require 'rails_helper'
```
Comment (polykoi) [inline spec/concepts/documentations/operation/show_spec.rb]: `spec/concepts/documentations/show_spec.rb`

File: spec/concepts/v1/lib/step/mailers/users/create_confirmation_spec.rb [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  describe '.call' do
+    before do
+      allow(V1::Lib::Service::EmailToken).to receive(:create)
+      allow(Rails).to receive_message_chain(:application, :config, :user_verification_path)
+      allow(V1::Users::EmailConfirmationWorker).to receive(:perform_async)
+    end
+
+    after { described_class.call(params, model: user) }
+
+    specify { expect(V1::Lib::Service::EmailToken).to receive(:create) }
```
Comment (polykoi) [inline spec/concepts/v1/lib/step/mailers/users/create_confirmation_spec.rb:16]: you can use `expect` alone

File: spec/concepts/v1/users/registrations/operation/create_spec.rb [hunk 7]
```diff
@@ -0,0 +1,61 @@
+require 'rails_helper'
```
Comment (polykoi) [inline spec/concepts/v1/users/registrations/operation/create_spec.rb]: `spec/concepts/v1/users/registrations/create_spec.rb`

File: spec/concepts/v1/users/registrations/operation/create_spec.rb [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+  end
+
+  describe 'Success' do
+    let(:params) { valid_params }
+
+    before do
+      allow(V1::Lib::Step::Mailers::Users::CreateConfirmation).to receive(:call).and_return(true)
+    end
+
+    it 'creates user' do
```
Comment (polykoi) [inline spec/concepts/v1/users/registrations/operation/create_spec.rb]: you can put all expectation in one example

File: spec/concepts/v1/users/registrations/operation/create_spec.rb [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+      expect { result }.to change(User, :count).from(0).to(1)
+      expect(result[:model]).to be_persisted
+      expect(result).to be_success
+    end
+
+    it 'calls shared step' do
+      expect(V1::Lib::Step::Mailers::Users::CreateConfirmation).to receive(:call)
```
Comment (polykoi) [inline spec/concepts/v1/users/registrations/operation/create_spec.rb]: you can use `expect` without `allow`

File: spec/concepts/v1/users/verifications/operation/confirm_spec.rb [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+  subject(:result) { described_class.call(params: params) }
+
+  let(:user) { create(:user) }
+  let(:valid_email_token) { valid_token_generator(:email_token, user.id) }
+  let(:valid_params) { { email_token: valid_email_token } }
+
+  describe 'Success' do
+    let(:params) { valid_params }
+
+    specify do
```
Comment (polykoi) [inline spec/concepts/v1/users/verifications/operation/confirm_spec.rb]: one example

File: spec/queries/v1/users/users_to_delete_query_spec.rb [hunk 11]
```diff
@@ -0,0 +1,27 @@
+require 'rails_helper'
+
+RSpec.describe V1::Users::UsersToDeleteQuery do
+  describe '.call' do
+    subject { described_class.call }
+
+    context 'users not found' do
+      specify { expect(subject).to be_empty }
```
Comment (polykoi) [inline spec/queries/v1/users/users_to_delete_query_spec.rb]: is_expected

File: spec/queries/v1/users/users_to_delete_query_spec.rb [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+    end
+
+    context 'users found' do
+      let(:not_confirmed_user) { create(:user) }
```
Comment (polykoi) [inline spec/queries/v1/users/users_to_delete_query_spec.rb:12]: let!

File: spec/requests/v1/users/verifications_spec.rb [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+          user.toggle!(:verified)
+          get confirmation_url
+        end
+
+        it 'returns error' do
+          expect(response).to be_unprocessable
+          expect(response).to match_json_schema('user/verification/confirm/errors')
+        end
+
+        it 'verificatioin fails', :dox do
```
Comment (polykoi) [inline spec/requests/v1/users/verifications_spec.rb]:
```suggestion
         it 'verification fails', :dox do
```

**Review by polykoi — CHANGES_REQUESTED** [review]

File: app/concepts/documentations/operation/show.rb (hunk 1 above)
Comment (bestwebua) [reply app/concepts/documentations/operation/show.rb:4]: What's wrong? :)

screenshot from 2018-12-24 15-57-22

**Pull request merged**